### DISCUSSION TITLE
Connection_user_constraint

### DISCUSSION THREAD

**Discussion by josefkoell** [spine-tools/SpineOpt.jl#573] [Support]
Hi!
I am currently working on an offshore energy model in SpineOpt, and I need help with connection__user_constraints. I want to set a constraint that if connection_A is used, nothing can flow through connection_B. Is there a similar parameter for connections as the units_on_coeffiecient parameter for unit__user_constraints? I would be very grateful for any help!

**Comment by DillonJ** [comment]
Hi @josefkoell 

Currently connections do not have an online variable. You have a couple of options to accomplish what you want.

The first is that you could simply represent your connections using units instead. 

The second way leverages the fact that if you create a unit not connected to any nodes, it will have only an online varable. You can use this in your user_constraints. You could create one of these switching units for each of your connections and use their online variables in the user_constraint. The nice things about this option is that you don't create any unecessary variables but you will need at least 3 user constraints (flow on connection 1 only if switching unit 1 is on, flow on connection 2 only if switching unit 2 is on and only switching unit 1 or switchng unit 2.

The third option invloves creating switching units in between your connection and the from node as below. The advantage of this is that you don't need user constraints but you create extra flow variables.

![image](https://user-images.githubusercontent.com/7080191/226687010-9a628615-38d0-4a41-a6cb-5d1f2027693d.png)

**Reply by josefkoell** [reply]
Thank you very much!
